### ISSUE TITLE
ECCurve.Validate() error message inconsistency for Explicit Curves

### ISSUE THREAD

**Issue by artemlos** [dotnet/corefx#19162]
When defining an explicit curve, there's an inconsistency in the error message.

If some of the parameters are incorrect, we get:
```
The specified prime curve parameters are not valid. Prime, A, B, G.X, G.Y and Order are required and must be the same length, and the same length as Q.X, Q.Y and D if those are specified. Seed, Cofactor and Hash are optional. Other parameters are not allowed.
```

Based on the message, only **Prime, A, B, G.X, G.Y and Order** are required and **Cofactor** is optional. However, based on [Validate() line L207](https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECCurve.cs#L207), **Cofactor** is a required parameter too.

**Comment by karelz** [comment]
That sounds reasonable. Do you plan to submit PR with a fix?

**Comment by artemlos** [comment]
I'm a bit unsure. Should the **cofactor** be a requirement for all explicit curves?

**Comment by karelz** [comment]
Isn't it what the code does? Or are there more code paths?

**Comment by danmosemsft** [comment]
@bartonjs make sense to update the message? seems to match the code.

**Comment by bartonjs** [comment]
@steveharter I don't suppose you remember why we have a mismatch between the code and the string here?  Did one of the platforms consider cofactor required but another didn't?

**Comment by steveharter** [comment]
IIRC OpenSsl requires the cofactor but Cng did not for explicit curves. The messages came when designing for Cng, and apparently it was not caught\changed when OpenSsl was added.

So yes the message should be updated.

p.s. according to the "SEC 1" spec in section C.2 "Syntax for Elliptical Curve Domain Parameters" cofactor is optional. However I suppose it could be considered required if someone wanted to verify the domain parameters as being correct (too expensive tho).

**Comment by bartonjs** [comment]
Yeah, looks like OpenSSL won't load an explicit key from file if Cofactor is missing.  (Or, at least, can't pretty print it back out).  So we should change the message for now.  If someone wants to come back and figure out how to make OpenSSL auto-calculate the cofactor, that can be done in a future version.
